prometheus-to-sd/main.go

```diff
@@ -111,6 +114,23 @@ func main() {
 	<-make(chan int)
 }
 
+func getSourceConfigs(gceConfig *config.GceConfig) (sourceConfigs []config.SourceConfig) {
+	if *asDaemonSet {
+		glog.Info("Taking source configs from api-server")
+		var err error
+		sourceConfigs, err = config.GetConfigsFromApiServer(gceConfig, *namespace)
```

Comment:
Pods are dynamic, what happens if some pod is added/removed. You should redesign whole pipeline to allow dynamic sources